sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/HDFSMetadataLogSuite.scala

```diff
@@ -209,14 +209,21 @@ class HDFSMetadataLogSuite extends SparkFunSuite with SharedSQLContext {
     }
 
     // Open and delete
-    val f1 = fm.open(path)
-    fm.delete(path)
-    assert(!fm.exists(path))
-    intercept[IOException] {
-      fm.open(path)
+    // Open and delete
+    if (Utils.isWindows) {
+      Utils.tryWithResource(fm.open(path))(_ => ())
+      fm.delete(path)
+      fm.delete(path) // should not throw exception
```

Comment:
Windows holds a exclusive lock so it does not allow to remove a file when it is open. So, I re-wrote the test cases.